frontend/components/puzzle4.jsx

```diff
@@ -70,17 +91,18 @@ class Puzzle4 extends React.Component {
             </p>
         }
         <KeypadContainer>
-          <Button click={this.state.value[0]} onClick={this.handleClick(0)} >1</Button>
-          <Button click={this.state.value[1]} onClick={this.handleClick(1)} >2</Button>
-          <Button click={this.state.value[2]} onClick={this.handleClick(2)} >3</Button>
-          <Button click={this.state.value[3]} onClick={this.handleClick(3)} >4</Button>
-          <Button click={this.state.value[4]} onClick={this.handleClick(4)} >5</Button>
-          <Button click={this.state.value[5]} onClick={this.handleClick(5)} disabled={puzzleBSolved}>6</Button>
-          <Button click={this.state.value[6]} onClick={this.handleClick(6)} >7</Button>
-          <Button click={this.state.value[7]} onClick={this.handleClick(7)} >8</Button>
-          <Button click={this.state.value[8]} onClick={this.handleClick(8)} >9</Button>
+          <Button click={this.state.value[0]} onClick={this.handleClick(0)} ><SafeImage src={images[0].src} ></SafeImage></Button>
```

Comment:
these can be self closing elements (`<SafeImage ... />`)

Reply:
also true statement